lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -310,6 +313,71 @@ sub settings {
     $self->render('test/settings');
 }
 
+=over 4
+
+=item show_filesrc()
+
+Returns the context of a config file of the selected job.
+So this works in the same way as the test module source.
+
+=back
+
+=cut
+
+sub show_filesrc {
+    my ($self)      = @_;
+    my $job         = $self->_stash_job or return $self->reply->not_found;
+    my $jobid       = $self->param('testid');
+    my $dir         = $self->stash('dir');
+    my $data_uri    = $self->stash('link_path');
+    my $testcasedir = testcasedir($job->DISTRI, $job->VERSION);
+    if (my $casedir = $job->settings->single({key => 'CASEDIR'})) {
+        my $casedir_url = Mojo::URL->new($casedir->value);
+        # if CASEDIR points to a remote location let's assume it is a git repo
+        # that we can reference like gitlab/github
+        last unless $casedir_url->scheme;
+        my $refspec = $casedir_url->fragment;
+        # try to read vars.json from resultdir and replace branch by actual git hash if possible
+        eval {
+            my $vars_json = Mojo::File->new($job->result_dir(), 'vars.json')->slurp;
+            my $vars      = decode_json($vars_json);
+            $refspec = $vars->{TEST_GIT_HASH};
```

Comment:
That assignment should only happen when `$vars->{TEST_GIT_HASH}` has a value, right?

Reply:
correct

Reply:
So it would still make sense to keep `$refspec` if `TEST_GIT_HASH` is not present in `vars.json`, e.g. because `isotovideo` failed early.

Reply:
xm! this block of code it is exactly the same as in Step.pm in src. From my tests, and from what i have checked on OSD the `TEST_GIT_HASH` should be assigned in the settings (dynamically) and be found in all of them. So my assumption is that we dont expect a miss here.

Reply:
I guess it doesn't matter that much. Note that the code in `Step.pm` simply might not have considered this as well. In your tests you've likely never had an early exit of isotovideo.